src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -391,7 +391,19 @@ export class SavedObjectsRepository {
         expectedResult.rawMigratedDoc._source
       );
 
-      return { tag: 'Right' as 'Right', value: expectedResult };
+      return {
+        tag: 'Right' as 'Right',
+        value: {
+          ...expectedResult,
+          rawMigratedDoc: {
+            ...expectedResult.rawMigratedDoc,
+            _source: {
+              ...expectedResult.rawMigratedDoc._source,
+              namespaces: savedObjectNamespaces ?? [getNamespaceString(savedObjectNamespace)],
+            },
+          },
+        },
+      };
```

Comment:
Since we're using the `this._rawToSavedObject` method below, I don't think this change is necessary, is it?

Reply:
hmm I seem to recall an earlier test failure, but I may be mis-remembering. Let me revert and see what happens here.

Reply:
update: you are absolutely right! The failure I was seeing was due to a setup assertion that needed fixing, rather than a legitimate test failure.